A few autopatching issues

drewmoseley opened the pull request:
These were discovered in my recent work on the i.MX7 Sabre port.  Note that this depends on #486 since I had to modify meta-mender-core/recipes-bsp/u-boot/patches/0002-Generic-boot-code-for-Mender.patch and getting things to apply cleanly didn't work as completely separate PRs.

drewmoseley commented:
@kacf note that https://github.com/mendersoftware/meta-mender/pull/488/commits/984869cd0a51d542834c5c3f0aa0eba106bf1f3d addresses the issue we discussed where the loadaddr is only set by env_default.h.

It's unclear to me if this fix will have adverse effects on other platforms or not.

kacf commented on meta-mender-core/recipes-bsp/u-boot/patches/0002-Generic-boot-code-for-Mender.patch:
```diff
@@ -143,8 +143,15 @@ index 0000000000..55f9bca7a1
++# define MENDER_DEFAULT_LOADADDR
++#else
++# define MENDER_DEFAULT_LOADADDR "kernel_addr_r=" __stringify(CONFIG_LOADADDR) "\0"
++#endif
```
Btw, I'm considering switching to `loadaddr` simply because it seems to be the preferred one in `env_default.h`. `kernel_addr_r` was somewhat arbitrarily chosen.

drewmoseley commented:
@kacf I removed the change for u-boot-fw-utils.  Looks like https://github.com/mendersoftware/meta-mender/pull/488/commits/dc79a314885abec492b005e5e9c5c3c2ec6a6c07 is a better fix.  Basically there is another mechanism for specifying UBOOT_MACHINE and with out that change, machines that use UBOOT_CONFIG instead will not have the UBOOT_MACHINE variable set without it.

Also, I reverted the sense of the conditional in my patch to u-boot-mender-common.inc

Regarding the race condition fixed here, I don't think it's related to that other ticket; that one appears to be something with bitbake metadata whereas this race condition would cause builds to fail later on due to files not being deployed yet.  I'll also comment as such in the ticket.  It is _possible_ that https://github.com/mendersoftware/meta-mender/pull/488/commits/dc79a314885abec492b005e5e9c5c3c2ec6a6c07 is related but I'm not certain.

drewmoseley commented:
Also note that I added an extra commit here.  https://github.com/mendersoftware/meta-mender/pull/488/commits/5e91cf71a162eb8b498288d114cfbb69db014b79

Not related to autopatching but something I noticed while nosing around the files today.

kacf commented:
This is still failing in Jenkins though. I'm planning to do a big pull request roundup in the next sprint, where I pick up several of the pull requests that have been problematic lately. I will leave this until then, but feel free to look at the error if you have any idea why.

drewmoseley commented:
This has all been subsumed into https://github.com/mendersoftware/meta-mender/pull/486.
